File: selectSort.cpp
#include <iostream>
#include <vector>

using namespace std;

void selectSort(vector<int>& arr) {
	int temp;
	for (int i = 0; i < arr.size(); ++i) {
		int index = i;
		for (int j = i + 1; j < arr.size(); ++j) {
			if (arr[index] > arr[j]) {
				index = j;
			}
		}
		if (index != i) {
			temp = arr[index];
			arr[index] = arr[i];
			arr[i] = temp;
		}
	}
}

int main() {
	int num;
	vector<int> nums;
	while (cin >> num) {
		nums.push_back(num);
		if (cin.get() == '\n') {
			break;
		}
	}
	for (int i = 0; i < nums.size(); ++i) {
		cout << nums[i] << ' ';
	}
	cout << endl;
	selectSort(nums);
	for (int i = 0; i < nums.size(); ++i) {
		cout << nums[i] << ' ';
	}
	return 0;
}

File: selectSort.js
// Select Sort
// select the minimum value in the array and put it to the head of the array
// best\Worst\Avg:O(n)
function selectSort(arr){
    const len = arr.length;
    // define a variable to store the minimum index of current interval
    let minIndex;
    // i is the start point of current interval
    for(let i = 0; i < len - 1; i++){
        // init minIndex 
        minIndex = i;
        // j is current interval's right limit
        for(let j = i; j < len; j++){
            // find the smallest data and update index of minIndex 
            if(arr[j] < arr[minIndex]){
                minIndex = j;
            }
        }
        // if minIndex is not current head index, swap them
        if(minIndex !== i){
            [arr[i], arr[minIndex]] = [arr[minIndex], arr[i]];
        }
    }
    return arr;
}